### PR TITLE
Add zfs localpv dashboard

### PR DESCRIPTION
**Pull request by rajaSahil**
### What this PR does?
- Add dashboard for zfs localpv
- Dashboard : https://raw.githubusercontent.com/openebs/zfs-localpv/develop/deploy/sample/grafana-dashboard.json

#### Screenshots:
Screenshot from 2021-09-16 14-15-32
Screenshot from 2021-09-16 16-19-10
Screenshot from 2021-09-16 16-19-18

### CODE REVIEW

File: jsonnet/config.libsonnet [hunk 1]
```diff
@@ -269,6 +271,22 @@
...
       },
+      zfsLocalPV: {
+        // To generate manifests for zfs localpv. If set, manifests will be generated.
+        enabled: true,
+        // To generate dashboards configMap yamls. If set, dashboards will be appended in grafana-dashboardDefinition yaml.
+        dashboards: true,
+        // To generate prometheusRule yamls. If set, prometheusRule will be generated.
+        alertRules: false,
+        // ServiceMonitor configuration
+        serviceMonitor: {
```
Comment (avishnu) [inline jsonnet/config.libsonnet:282]: Should this be true or false?

File: jsonnet/openebs-mixin/dashboards/openebs/zfslocalpv.json [hunk 2]
```diff
@@ -0,0 +1,1245 @@
+{
```
Comment (avishnu) [inline jsonnet/openebs-mixin/dashboards/openebs/zfslocalpv.json:1]: Just recollecting.. why do we need to check-in the same file twice?

File: jsonnet/config.libsonnet (hunk 1 above)
Comment (rajaSahil) [reply jsonnet/config.libsonnet:282]: There are no alert rules, serviceMonitor and podMonitor for zfs localpv. Therefore it is set as false.
The dashboard is built using node-exporter metrics.

File: jsonnet/openebs-mixin/dashboards/openebs/zfslocalpv.json (hunk 2 above)
Comment (rajaSahil) [reply jsonnet/openebs-mixin/dashboards/openebs/zfslocalpv.json:1]: To include the dashboard in helm installation, we are generating a dashboard using mixin.

File: deploy/charts/values.yaml [hunk 3]
```diff
@@ -648,3 +648,8 @@ openebsMonitoringAddon:
     enabled: true
     dashboards:
       enabled: true
+
+  zfsLocalPV:
```
Comment (Ab-hishek) [inline deploy/charts/values.yaml:652]: Can you update the charts README accordingly?

File: deploy/charts/dashboards/zfsLocalPV/zfslocalpv.json [hunk 4]
```diff
@@ -0,0 +1,1258 @@
...
+            "description": null,
+            "error": null,
+            "hide": 0,
+            "includeAll": false,
+            "label": null,
+            "multi": false,
+            "name": "node",
+            "options": [ ],
+            "query": {
+               "query": "label_values(node_exporter_build_info{job=~\"node-exporter\"}, instance)",
```
Comment (Ab-hishek) [inline deploy/charts/dashboards/zfsLocalPV/zfslocalpv.json]: Will this nested query not cause issue with the old grafana versions like we faced for lvm localpv?

File: deploy/charts/dashboards/zfsLocalPV/zfslocalpv.json [hunk 5]
```diff
@@ -0,0 +1,1258 @@
...
+               "refId": "Prometheus-node-Variable-Query"
+            },
+            "refresh": 1,
+            "regex": "/([^:]+):.*/",
+            "skipUrlSync": false,
+            "sort": 1,
+            "tagValuesQuery": "",
+            "tags": [ ],
```
Comment (Ab-hishek) [inline deploy/charts/dashboards/zfsLocalPV/zfslocalpv.json:1191]: We can add tags for `OpenEBS` and `zfs-localpv`.

File: deploy/charts/dashboards/zfsLocalPV/zfslocalpv.json [hunk 6]
```diff
@@ -0,0 +1,1258 @@
...
+         "title": "ARC L2",
+         "type": "row"
+      }
+   ],
+   "refresh": "10s",
+   "schemaVersion": 27,
+   "style": "dark",
+   "tags": [
+      "OpenEBS",
+      "ZFS-LocalPV"
```
Comment (Ab-hishek) [inline deploy/charts/dashboards/zfsLocalPV/zfslocalpv.json:1146]: Should this be in lower case or upper case?

**Comment by Ab-hishek** [comment]
In the `Used Space` panel, I see repeating legends. Is this the expected visualization? IMO, I feel better to have legends in table format with the avg % shown.

**Comment by rajaSahil** [comment]
I haven't made any changes to the dashboard panels or legends. Will update the legends in table format.

**Comment by rajaSahil** [comment]
@Ab-hishek repeated legends were being shown because of my setup. Have updated the dashboard screenshots.

File: deploy/charts/dashboards/zfsLocalPV/zfslocalpv.json (hunk 6 above)
Comment (rajaSahil) [reply deploy/charts/dashboards/zfsLocalPV/zfslocalpv.json:1146]: Not sure. For `lvm` pool dashboard, we have used uppercase.

**Review by kmova — APPROVED** [review]

**Review by avishnu — APPROVED** [review]
Lgtm

**Review by Ab-hishek — APPROVED** [review]

**Pull request merged**